app/src/main/java/ch/hikemate/app/ui/auth/SignInScreen.kt

```diff
@@ -83,83 +85,93 @@ fun SignInScreen(
             Log.d("MainActivity", "addAccountLauncher result: $result")
           }
 
+  val errorMessageIdState = authViewModel.errorMessageId.collectAsState()
+  val loadingState = authViewModel.loading.collectAsState()
+
   // If the user is already signed in, navigate to the map screen
   LaunchedEffect(authViewModel.currentUser.collectAsState().value) {
     if (authViewModel.isUserLoggedIn()) {
       navigationActions.navigateTo(TopLevelDestinations.MAP)
     }
   }
 
-  Scaffold(
-      modifier = Modifier.fillMaxSize().testTag(Screen.AUTH),
-      content = { padding ->
-        Box(
-            modifier = Modifier.fillMaxSize(),
-        ) {
-          // I created a box to make the background image take a little more space
-          // than the screen size, so that the blur effect doesn't show white edges
+  AsyncStateHandler(
+      errorMessageIdState = errorMessageIdState,
+      actionContentDescriptionStringId = R.string.retry,
+      actionOnErrorAction = { navigationActions.navigateTo(Route.AUTH) },
+      loadingState = loadingState,
+  ) {
+    Scaffold(
+        modifier = Modifier.fillMaxSize().testTag(Screen.AUTH),
+        content = { padding ->
           Box(
-              modifier = Modifier.fillMaxSize().clipToBounds().scale(1.1f),
-          ) {
-            Image(
-                painter = painterResource(id = R.drawable.sign_in_background),
-                contentDescription = "Background Image",
-                modifier = Modifier.fillMaxSize().blur(10.dp),
-                contentScale = ContentScale.Crop,
-            )
-          }
-          Column(
-              modifier = Modifier.fillMaxSize().padding(padding),
-              horizontalAlignment = Alignment.CenterHorizontally,
-              verticalArrangement = Arrangement.SpaceAround,
+              modifier = Modifier.fillMaxSize(),
           ) {
-            Column(
-                horizontalAlignment = Alignment.CenterHorizontally,
+            // I created a box to make the background image take a little more space
+            // than the screen size, so that the blur effect doesn't show white edges
+            Box(
+                modifier = Modifier.fillMaxSize().clipToBounds().scale(1.1f),
```

Comment:
Question: Why do you need a Box in a Box? Couldn't you replace one of it with a Column or a Row?

Reply:
I didn't do that part of the code. It was already there.

Reply:
Oh okay, it will need to be updated soon in another PR